schemas/JSON/configuration/configuration.schema.0.1.json

```diff
@@ -0,0 +1,204 @@
+                        },
+                        "version": {
+                            "$ref": "#/def/version"
+                        },
+                        "minimumVersion": {
+                            "$ref": "#/def/version"
+                        },
+                        "productId": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The product id."
+                        },
+                        "channelId": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The channel id."
+                        },
+                        "publisher": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The name of the publisher."
+                        },
+                        "vsconfigFile": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The VS config file."
+                        },
+                        "os": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The operating system."
+                        },
+                        "vaultType": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The key vault type."
+                        },
+                        "vaultName": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The name of the key vault."
+                        },
+                        "apikey": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The api key."
+                        },
+                        "repository": {
+                            "type": "string",
+                            "maxLength": 128,
+                            "description": "The name of the repository."
+                        },
+                        "extensions": {
+                            "type": [ "string", "null" ],
+                            "description": "The list of extensions",
+                            "uniqueItems": true
+                        },
+                        "windowsUpdate": {
+                            "type": [ "boolean", "null" ],
+                            "description": "Indicates whether to allow Windows to update."
+                        }
+                    }
+                },
+                "directives": {
+                    "type": "object", 
+                    "properties": {

```

Comment:
Is the plan to be strict on directives?  That does provide the possibility of the guarantee that "if you can provide it, it will be implemented".  I think that if we want to be strict here, it might be good to provide a defined extensibility point for users to attach arbitrary data to a resource.

Reply:
I set the ```"additionalProperties": true``` so that arbitrary data can be included in the ```directives``` object. I left the properties as is for now since I'm not sure which ones are confirmed yet.